### PR TITLE
:wrench: Infra/editorconfig and eslint-import

### PR DESCRIPTION
**Pull request by tuhacrt**
## Describe your changes

1. (root): add `.editorconfig` to setup `indent_size` and `end_of_line` instead of `.vscode` `settings.json` (for compatibility of other IDEs like `vim`, `nvim`, `webstorm`...).
2. (packages/domain_layer): add `eslint-plugin-import` to proper separate import by scopes.
3. (root): revise package.json to suit npm package rules. (`Kakerlaken-Poker` => `kakerlaken-poker`), and lint using [sort-package-json](https://www.npmjs.com/package/sort-package-json).

---

## What is the scope of this change?
- [x] **Root** // I think there should has a `Root` scope for `ci` or dependency change.

backend
- [x] **Domain**
- [ ] **Repository**
- [ ] **UseCase**
- [ ] **Api**

frontend
- [ ] **Ui**
- [ ] **Web**

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -2,14 +2,13 @@
...
+    "preinstall",
     "swcrc",
     "Turborepo",
     "typecheck",
     "unplugin"
   ],
   "editor.formatOnPaste": true,
   "editor.formatOnSave": true,
-  "editor.formatOnType": true,
-  "editor.tabSize": 2,
```
Comment (Dlutermade) [inline .vscode/settings.json:13]: 這需要

File: .vscode/settings.json [hunk 2]
```diff
@@ -2,14 +2,13 @@
...
-  "prettier.endOfLine": "lf"
```
Comment (Dlutermade) [inline .vscode/settings.json:14]: 這也需要

File: packages/domain_layer/.eslintrc.js [hunk 3]
```diff
@@ -22,5 +22,20 @@ module.exports = {
     '@typescript-eslint/explicit-function-return-type': 'off',
     '@typescript-eslint/explicit-module-boundary-types': 'off',
     '@typescript-eslint/no-explicit-any': 'off',
+    'import/order': [
```
Comment (Dlutermade) [inline packages/domain_layer/.eslintrc.js:25]: 這邊是幹嘛的 ~

File: .editorconfig [hunk 4]
```diff
@@ -0,0 +1,15 @@
+# http://editorconfig.org
```
Comment (Dlutermade) [inline .editorconfig:1]: 這是什麼東西~

**Comment by Dlutermade** [comment]
eslint-plugin-import 這是什麼阿第一次看到~

**Review by Dlutermade — APPROVED** [review]

File: .vscode/settings.json (hunk 1 above)
Comment (tuhacrt) [reply .vscode/settings.json:13]: 笑死，怎麼問完我還沒回就 merge 了

File: packages/domain_layer/.eslintrc.js (hunk 3 above)
Comment (tuhacrt) [reply packages/domain_layer/.eslintrc.js:25]: 這邊是讓 import 根據不同的 scope 來自動排序，以下示範。

```typescript
// no import/order
import { createDeck } from './createDeck';

import { pipe } from 'fp-ts/function';
import { Creature } from 'card';
import { readFile } from 'fs';
```

```typescript
// with import/order and 'newlines-between'
import { readFile } from 'fs'; // builtin scope from 'node'

import { Creature } from 'card'; // external scope
import { pipe } from 'fp-ts/function';

import { createDeck } from './createDeck'; // sibling scope
```

reference: 
[sort-imports](https://eslint.org/docs/latest/rules/sort-imports)

**Pull request merged**